packages/protocol/contracts/stability/Exchange.sol

```diff
@@ -273,17 +268,20 @@ contract Exchange is IExchange, Initializable, Ownable, UsingRegistry {
     uint256 buyTokenBucket;
     (buyTokenBucket, sellTokenBucket) = _getBuyAndSellBuckets(sellGold);
 
-    uint256 x = spread.denominator.sub(spread.numerator).mul(sellAmount);
-    uint256 numerator = x.mul(buyTokenBucket);
-    uint256 denominator = sellTokenBucket.mul(spread.denominator).add(x);
+    int256 reducedSellAmount = FixidityLib.fixed1()
+      .subtract(spread)
+      .multiply(FixidityLib.newFixed(int256(sellAmount)));
```

Comment:
should we make helper functions for mixed arithmetic operations?

Reply:
I think it would be nice to have an inlined library function that does `newFixed` conversion and division, like `divideFixed(uint256, uint256)` which converts both to fixed and divides, returning an `int256`. I think a complement function `1-x` would also be useful.